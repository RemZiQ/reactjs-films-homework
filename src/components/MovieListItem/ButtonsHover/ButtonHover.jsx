import React from 'react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';


import './buttonsHover.scss';

const ButtonHover = ({
  id,
  handleOpenOverview,
  fetchTrailer,
  handleShowTrailer,
  location,
  fetchMovie,
}) => {
  const handle = (e) => {
    handleShowTrailer();
    try {
      fetchTrailer(e);
    } catch (error) {
      console.log(error);
    }
  };

  const handleOpenOverviewAndFetchCurrentMovie = (e) => {
    handleOpenOverview();
    fetchMovie(e.target.id);
  };

  const param = new URLSearchParams(location.search);
  const query = param.get('query');
  const currentQuery = query === null ? 'init793797112020979' : query;

  return (
    <div className="movieList__item_buttons_container" id={`${id}`}>
      <div className="movieList__item__button_watch_container">
        <button
          onClick={handle}
          type="button"
          className="movieList__item__button_watch"
          id={id}
        />
        <p>Watch now</p>
      </div>
      <Link to={{ pathname: '/search', search: `?query=${currentQuery}&id=${id}` }}>
        <button
          onClick={handleOpenOverviewAndFetchCurrentMovie}
          type="button"
          className="movieList__item__button_view"
          id={`${id}`}
        >
        View info
        </button>
      </Link>
    </div>
  );
};

ButtonHover.propTypes = {
  id: propTypes.number,
  handleOpenOverview: propTypes.func,
  fetchTrailer: propTypes.func,
  handleShowTrailer: propTypes.func,
  location: propTypes.objectOf(propTypes.any),
  fetchMovie: propTypes.func,
};

ButtonHover.defaultProps = {
  id: null,
  handleOpenOverview: null,
  fetchTrailer: null,
  handleShowTrailer: null,
  location: null,
  fetchMovie: null,
};

export default ButtonHover;
