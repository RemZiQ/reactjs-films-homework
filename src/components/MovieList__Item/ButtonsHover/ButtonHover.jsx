import React, { Component } from 'react';

import './buttonsHover.scss';


// class ButtonHover extends Component{
//   constructor(props) {
//     super(props);
//     // this.addClass = this.addClass.bind(this);
//   }
//   // addClass(){
//   //   console.log(this.buttonView);
//   //   this.buttonView.classList.add('movieList__item_bgOverwiew');
//   // }
//   render(){
//     return (
//       <div className='movieList__item_buttons_container' id={`${this.props.id}`}>
//         <div className='movieList__item__button_watch_container'>
//           <button type='button' className='movieList__item__button_watch' id={this.props.id}></button>
//           <p>Watch now</p>
//         </div>
//         <button ref={(button) => {this.buttonView = button}} onClick={this.props.handle}
//          type='button' className='movieList__item__button_view'>View info</button>
//       </div>
//     )
//   }
// }
const ButtonHover = ({
  id, handleOpenOverview, fetchTrailer, handleShowTrailer, state,
}) => (
  <div className="movieList__item_buttons_container" id={`${id}`}>
    <div className="movieList__item__button_watch_container">
      <button
        onClick={fetchTrailer}
        onClick={handleShowTrailer}
        type="button"
        className="movieList__item__button_watch"
        id={id}
      />
      <p>Watch now</p>
    </div>
    <button onClick={handleOpenOverview} type="button" className="movieList__item__button_view">View info</button>
  </div>
);

export default ButtonHover;
