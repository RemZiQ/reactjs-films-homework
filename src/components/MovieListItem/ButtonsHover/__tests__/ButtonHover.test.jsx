import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { ButtonHover } from '../ButtonHover';


it('ButtonHover renders well with query', () => {
  const renderer = new ShallowRenderer();
  const props = {
    location: { search: '?query=red' },
  };
  const result = renderer.render(<ButtonHover {...props} />);
  expect(result).toMatchSnapshot();
});

it('ButtonHover renders well without query', () => {
  const renderer = new ShallowRenderer();
  const props = {
    location: { search: '' },
  };
  const result = renderer.render(<ButtonHover {...props} />);
  expect(result).toMatchSnapshot();
});
