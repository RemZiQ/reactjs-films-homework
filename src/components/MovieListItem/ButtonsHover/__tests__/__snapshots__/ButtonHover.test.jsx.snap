// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonHover renders well with query 1`] = `
<div
  className="movieList__item_buttons_container"
  id="null"
>
  <div
    className="movieList__item__button_watch_container"
  >
    <button
      className="movieList__item__button_watch"
      id={null}
      onClick={[Function]}
      type="button"
    />
    <p>
      Watch now
    </p>
  </div>
  <Link
    to={
      Object {
        "pathname": "/search",
        "search": "?query=red&id=null",
      }
    }
  >
    <button
      className="movieList__item__button_view"
      id="null"
      onClick={[Function]}
      type="button"
    >
      View info
    </button>
  </Link>
</div>
`;

exports[`ButtonHover renders well without query 1`] = `
<div
  className="movieList__item_buttons_container"
  id="null"
>
  <div
    className="movieList__item__button_watch_container"
  >
    <button
      className="movieList__item__button_watch"
      id={null}
      onClick={[Function]}
      type="button"
    />
    <p>
      Watch now
    </p>
  </div>
  <Link
    to={
      Object {
        "pathname": "/search",
        "search": "?query=init793797112020979&id=null",
      }
    }
  >
    <button
      className="movieList__item__button_view"
      id="null"
      onClick={[Function]}
      type="button"
    >
      View info
    </button>
  </Link>
</div>
`;
